pkgs/development/compilers/rust/binaryBuild.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv, fetchurl, makeWrapper, cacert, zlib, buildRustPackage, curl
+, version
+, src
+, versionType
+, bootstrapping
+}:
+
+let
+  inherit (stdenv.lib) optionalString;
+
+  platform = if stdenv.system == "x86_64-linux"
```

Comment:
I'll remove the duplication.

I claim that adding binary nightly builds for other platforms is outside the scope of this diff (they don't exist before this diff either).